server/main.go

```diff
@@ -3,166 +3,177 @@ package main
 import (
 	"bufio"
 	"encoding/gob"
+	scs "github.com/cpalsulich/simple-chat-server"
 	"log"
 	"net"
-	scs "simple-chat-server"
 )
 
 func main() {
 	clients := make(map[string]*scs.User)
 	rooms := make(map[string]*scs.Room)
-	err := server(clients, rooms)
 
-	if err != nil {
+	if err := startServer(clients, rooms); err != nil {
 		panic(err)
 	}
 }
 
-func server(users map[string]*scs.User, rooms map[string]*scs.Room) error {
+type server struct {
+	rooms      map[string]*scs.Room
+	users      map[string]*scs.User
+	creating   chan scs.Room
+	connecting chan scs.User
```

Comment:
Does this need to keep track of users disconnecting?

Reply:
Probably